File: AgenteGlobal/AgenteGlobal/calculadora.py
#!/usr/bin/env python3
"""
Calculadora CLI interativa
Operações: soma, subtração, multiplicação, divisão, potência e raiz quadrada.
"""

import math
import sys


# ---------------------------------------------------------------------------
# Funções de operação
# ---------------------------------------------------------------------------

def somar(a: float, b: float) -> float:
    """Retorna a soma de a e b."""
    return a + b


def subtrair(a: float, b: float) -> float:
    """Retorna a subtração de a por b."""
    return a - b


def multiplicar(a: float, b: float) -> float:
    """Retorna a multiplicação de a por b."""
    return a * b


def dividir(a: float, b: float) -> float:
    """Retorna a divisão de a por b. Levanta ValueError se b for zero."""
    if b == 0:
        raise ValueError("Divisão por zero não é permitida.")
    return a / b


def potenciar(base: float, expoente: float) -> float:
    """Retorna base elevada ao expoente."""
    return base ** expoente


def raiz_quadrada(valor: float) -> float:
    """Retorna a raiz quadrada de valor. Levanta ValueError se valor for negativo."""
    if valor < 0:
        raise ValueError("Raiz quadrada de número negativo não é permitida (números reais).")
    return math.sqrt(valor)


# ---------------------------------------------------------------------------
# Utilidades de entrada
# ---------------------------------------------------------------------------

def ler_numero(prompt: str) -> float:
    """
    Lê um número float do stdin.
    Repete até que o usuário digite um valor válido.
    """
    while True:
        try:
            return float(input(prompt))
        except ValueError:
            print("  ⚠ Entrada inválida. Digite um número válido (ex.: 3.14).")


def exibir_menu() -> None:
    """Imprime o menu de operações."""
    print("\n" + "=" * 40)
    print("       CALCULADORA EM PYTHON")
    print("=" * 40)
    print("  1) Somar            (a + b)")
    print("  2) Subtrair         (a - b)")
    print("  3) Multiplicar      (a * b)")
    print("  4) Dividir          (a / b)")
    print("  5) Potência         (a ^ b)")
    print("  6) Raiz quadrada    (√a)")
    print("  0) Sair")
    print("=" * 40)


# ---------------------------------------------------------------------------
# Loop principal
# ---------------------------------------------------------------------------

def main() -> None:
    """Loop principal da calculadora CLI."""
    while True:
        exibir_menu()
        escolha = input("Escolha uma opção: ").strip()

        # --- Sair ---------------------------------------------------------
        if escolha == "0":
            print("Encerrando a calculadora. Até logo!")
            break

        # --- Operações binárias -------------------------------------------
        if escolha in ("1", "2", "3", "4", "5"):
            a = ler_numero("  Digite o primeiro número (a): ")
            b = ler_numero("  Digite o segundo número (b): ")

            try:
                if escolha == "1":
                    resultado = somar(a, b)
                    operacao = f"{a} + {b}"
                elif escolha == "2":
                    resultado = subtrair(a, b)
                    operacao = f"{a} - {b}"
                elif escolha == "3":
                    resultado = multiplicar(a, b)
                    operacao = f"{a} * {b}"
                elif escolha == "4":
                    resultado = dividir(a, b)
                    operacao = f"{a} / {b}"
                elif escolha == "5":
                    resultado = potenciar(a, b)
                    operacao = f"{a} ^ {b}"

                print(f"\n  ✅ Resultado: {operacao} = {resultado}")

            except ValueError as e:
                print(f"\n  ❌ Erro: {e}")

        # --- Raiz quadrada (unária) --------------------------------------
        elif escolha == "6":
            a = ler_numero("  Digite o número (a): ")
            try:
                resultado = raiz_quadrada(a)
                print(f"\n  ✅ Resultado: √{a} = {resultado}")
            except ValueError as e:
                print(f"\n  ❌ Erro: {e}")

        # --- Opção inválida ----------------------------------------------
        else:
            print("\n  ⚠ Opção inválida. Escolha um número entre 0 e 6.")


if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("\n\nCalculadora interrompida pelo usuário. Até logo!")
        sys.exit(0)

File: AgenteGlobal/AgenteGlobal/calculadora.js
#!/usr/bin/env node
'use strict';

/**
 * Calculadora CLI interativa em Node.js
 * Operações: soma, subtração, multiplicação, divisão,
 *            potência e raiz quadrada.
 * Usa o módulo readline nativo do Node.js.
 */

const readline = require('readline');

// ──────────────────────────────────────────────
// Funções de operação
// ──────────────────────────────────────────────

function soma(a, b) {
  return a + b;
}

function subtracao(a, b) {
  return a - b;
}

function multiplicacao(a, b) {
  return a * b;
}

function divisao(a, b) {
  if (b === 0) {
    throw new Error('Divisão por zero não é permitida.');
  }
  return a / b;
}

function potencia(base, expoente) {
  return Math.pow(base, expoente);
}

function raizQuadrada(valor) {
  if (valor < 0) {
    throw new Error('Não existe raiz quadrada real de número negativo.');
  }
  return Math.sqrt(valor);
}

// ──────────────────────────────────────────────
// Utilitários de entrada
// ──────────────────────────────────────────────

/**
 * Pergunta ao usuário e retorna uma Promise com a resposta.
 */
function perguntar(rl, texto) {
  return new Promise((resolve) => {
    rl.question(texto, (resposta) => resolve(resposta.trim()));
  });
}

/**
 * Solicita um número válido ao usuário.
 * Repete até receber um valor numérico válido.
 */
async function lerNumero(rl, prompt) {
  while (true) {
    const entrada = await perguntar(rl, prompt);
    const valor = Number(entrada);
    if (entrada !== '' && !Number.isNaN(valor) && Number.isFinite(valor)) {
      return valor;
    }
    console.log('  ⚠ Entrada inválida. Digite um número válido.');
  }
}

// ──────────────────────────────────────────────
// Menu
// ──────────────────────────────────────────────

function exibirMenu() {
  console.log('\n════════════════════════════════════════');
  console.log('          CALCULADORA CLI');
  console.log('════════════════════════════════════════');
  console.log('  1) Soma              (a + b)');
  console.log('  2) Subtração         (a - b)');
  console.log('  3) Multiplicação     (a * b)');
  console.log('  4) Divisão           (a / b)');
  console.log('  5) Potência          (a ^ b)');
  console.log('  6) Raiz Quadrada     (√a)');
  console.log('  0) Sair');
  console.log('════════════════════════════════════════\n');
}

// ──────────────────────────────────────────────
// Execução de operações
// ──────────────────────────────────────────────

async function executarOperacao(rl, opcao) {
  let a, b, resultado, simbolo;

  try {
    switch (opcao) {
      case '1':
        a = await lerNumero(rl, 'Digite o primeiro valor (a): ');
        b = await lerNumero(rl, 'Digite o segundo valor  (b): ');
        resultado = soma(a, b);
        simbolo = '+';
        break;

      case '2':
        a = await lerNumero(rl, 'Digite o primeiro valor (a): ');
        b = await lerNumero(rl, 'Digite o segundo valor  (b): ');
        resultado = subtracao(a, b);
        simbolo = '-';
        break;

      case '3':
        a = await lerNumero(rl, 'Digite o primeiro valor (a): ');
        b = await lerNumero(rl, 'Digite o segundo valor  (b): ');
        resultado = multiplicacao(a, b);
        simbolo = '*';
        break;

      case '4':
        a = await lerNumero(rl, 'Digite o primeiro valor (a): ');
        b = await lerNumero(rl, 'Digite o segundo valor  (b): ');
        resultado = divisao(a, b);
        simbolo = '/';
        break;

      case '5':
        a = await lerNumero(rl, 'Digite a base: ');
        b = await lerNumero(rl, 'Digite o expoente: ');
        resultado = potencia(a, b);
        simbolo = '^';
        break;

      case '6':
        a = await lerNumero(rl, 'Digite o valor: ');
        resultado = raizQuadrada(a);
        simbolo = '√';
        break;

      default:
        console.log('  ⚠ Opção inválida. Escolha uma opção de 0 a 6.');
        return;
    }

    // Exibe resultado
    if (opcao === '6') {
      console.log(`\n  ✓ √${a} = ${resultado}`);
    } else {
      console.log(`\n  ✓ ${a} ${simbolo} ${b} = ${resultado}`);
    }
  } catch (erro) {
    console.log(`\n  ✗ Erro: ${erro.message}`);
  }
}

// ──────────────────────────────────────────────
// Loop principal
// ──────────────────────────────────────────────

async function loopPrincipal() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  console.log('\nBem-vindo à Calculadora CLI! Use Ctrl+C para sair a qualquer momento.');

  while (true) {
    exibirMenu();
    const opcao = await perguntar(rl, 'Escolha uma opção: ');

    if (opcao === '0' || opcao.toLowerCase() === 'sair') {
      console.log('\nAté logo! 👋\n');
      rl.close();
      break;
    }

    await executarOperacao(rl, opcao);

    // Pausa antes de voltar ao menu
    await perguntar(rl, '\nPressione ENTER para continuar...');
  }
}

// ──────────────────────────────────────────────
// Ponto de entrada
// ──────────────────────────────────────────────

loopPrincipal().catch((erro) => {
  console.error('Erro inesperado:', erro);
  process.exit(1);
});